server/webhook.go

```diff
@@ -310,7 +294,43 @@ func (s *Server) createSchemaUpdateIssue(ctx context.Context, repository *api.Re
 
 	issue, err := s.createIssue(ctx, issueCreate, api.SystemBotID)
 	if err != nil {
-		s.l.Warn("Failed to create update schema task for added repository file", zap.Error(err),
+		s.l.Warn("Failed to create update schema issue for added repository file", zap.Error(err),
+			zap.String("file", added))
+		return nil, nil
+	}
+	return issue, nil
+}
+
+func (s *Server) createTenantSchemaUpdateIssue(ctx context.Context, repository *api.Repository, mi *db.MigrationInfo, vcsPushEvent vcs.VCSPushEvent, commit gitlab.WebhookCommit, added string, statement string) (*api.Issue, error) {
+	if mi.Environment != "" {
+		return nil, fmt.Errorf("environment isn't accepted in schema update for tenant mode project")
+	}
+
+	m := &api.UpdateSchemaContext{
+		MigrationType: mi.Type,
+		VCSPushEvent:  &vcsPushEvent,
+		UpdateSchemaDetailList: []*api.UpdateSchemaDetail{
+			{
+				DatabaseName: mi.Database,
+				Statement:    statement,
+			},
+		},
+	}
+	createContext, err := json.Marshal(m)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to construct issue create context payload, error %v", err)
+	}
+	issueCreate := &api.IssueCreate{
```

Comment:
What if we create 2 separate methods  buildUpdateSchemaContext and buildTenantUpdateSchemaContext?

And then we don't need to repeat L323 ~ 336

Reply:
As a 2nd thought, I feel separating at buildUpdateSchemaContext is more appropriate because then if buildUpdateSchemaContext returns error, we can emit the ignore warning and continue. While if createIssue returns error, it's likely an serious issue and we should just abort.

Putting them together in a single method makes caller hard to distinguish the difference.